async/logic/Async4BackendRest_DestinationAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,543 @@
+
+		for (ConsumerRecord<String, String> record : records) {
+			System.out.println(String.format("*record* %s:%s", record.offset(), record.value()));
+			receiveMessage = record.value();
+		}
+
+		// メッセージが拾えなかったら
+		if(receiveMessage == null) {
+			System.out.println("Async4BackendRest_DestinationAPI_Topic relode");
+			//continue;
+			return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+					}
+
+		Response typeresponse = ExecuteDestinationAPI(receiveMessage, destinationStatus, destinationResponseBody, executionTimes);
+
+		// 代理実行したAPIがエラーの場合
+		if (Integer.parseInt(destinationStatus.toString()) >= Response.Status.MOVED_PERMANENTLY.getStatusCode()) {
+			System.out.println("Destination API Error.");
+			// 実行回数超過の場合
+			if(3 < Integer.parseInt(executionTimes.toString())) {
+				// 超過
+				System.out.println("over 3times");
+
+			} else {
+				// メッセージ再生成(実行回数に1加算する)
+				typeresponse = CreateRetryMessage(receiveMessage, retryMessage, executionTimes);
+					
+				producer.beginTransaction();
+				
+				// Topic１に再登録
+				producer.send(new ProducerRecord<String, String>(INPUT_TOPIC, retryMessage.toString()));
+							
+				Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>();
+
+				for (TopicPartition partition : records.partitions()) {
+					List<ConsumerRecord<String, String>> partitionedRecords = records.records(partition);
+					long offset = partitionedRecords.get(partitionedRecords.size() - 1).offset();
+
+					offsetsToCommit.put(partition, new OffsetAndMetadata(offset + 1));
+				}
+
+				producer.sendOffsetsToTransaction(offsetsToCommit, CONSUMER_GROUP_ID);
+
+				// トランザクション確定
+				producer.commitTransaction();
+							
+				//continue;
+				return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+			}
+		}
+		
+		// 送信メッセージ生成
+		typeresponse = CreateSendMessage(receiveMessage, sendMessage, destinationStatus, destinationResponseBody);
+		
+		producer.beginTransaction();
+		
+		// メッセージをTopic2に送る
+		producer.send(new ProducerRecord<String, String>(OUTPUT_TOPIC, sendMessage.toString()));
+		
+		Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>();
+
+		for (TopicPartition partition : records.partitions()) {
+			List<ConsumerRecord<String, String>> partitionedRecords = records.records(partition);
+			long offset = partitionedRecords.get(partitionedRecords.size() - 1).offset();
+
+			offsetsToCommit.put(partition, new OffsetAndMetadata(offset + 1));
+		}
+		
+		producer.sendOffsetsToTransaction(offsetsToCommit, CONSUMER_GROUP_ID);
+		
+		// トランザクション確定
+		producer.commitTransaction();
+		
+
+		return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+	}
+
+	private Response ExecuteDestinationAPI(String receivemessage, StringBuilder destinationStatus, StringBuilder destinationResponseBody, StringBuilder executionTimes) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.